toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/AuthenticatorState.kt

```diff
@@ -125,6 +134,16 @@ public sealed interface AuthenticatorState : NetworkAuthenticationChallengeHandl
      * @since 200.2.0
      */
     public fun dismissAll()
+
+    /**
+     * Revokes OAuth tokens and all credentials from the AuthenticationManager.arcGISCredentialStore`
+     * and `AuthenticationManager.networkCredentialStore`. If an IAP credential exists in
+     * the `AuthenticationManager.arcGISCredentialStore`, this function will launch a custom tab for the user to invalidate
+     * the IAP session.
+     *
+     * @since 200.8.0
+     */
+    public suspend fun signOut()
```

Comment:
Since this is a suspend function, we should return `Result` here:
```suggestion
    public suspend fun signOut(): Result<Boolean>
```